File: Task_4/nomeer_ws/src/cmplx_num_pkg/src/pub_cmplx.cpp
#include <ros/ros.h>
#include <cmplx_num_pkg/cmplx.h>
#include <cstdlib>
int main(int argc, char** argv)
{
    // node & topic
    ros::init(argc, argv, "pub_cmplx_cpp");
    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<cmplx_num_pkg::cmplx>("cmplx_topic_cpp",1);
    ros::Rate rate(1);

    // message
    cmplx_num_pkg::cmplx cmplxNum;
    
    while (ros::ok())
    {
        srand(time(0));
        cmplxNum.real = ( (float)rand() / RAND_MAX) * 100;
        cmplxNum.imaginary = ( (float)rand() / RAND_MAX) * 100;
        pub.publish(cmplxNum);
        ros::spinOnce();
        rate.sleep();        
    }


    return 0;
}

File: Task_4/nomeer_ws/src/cmplx_num_pkg/scripts/pub_cmplx.py
#!/usr/bin/python3

import rospy
from cmplx_num_pkg.msg import cmplx
import random

def pub_cmplx():
    # node & topic
    rospy.init_node("pub_cmplx")
    pub = rospy.Publisher("cmplx_topic", cmplx, queue_size=1)
    rate = rospy.Rate(1)

    # message
    cmplxNum = cmplx()

    # publishing
    while not rospy.is_shutdown():
        cmplxNum.real = round(random.random() * 100,3)
        cmplxNum.imaginary = round(random.random() * 100,3)

        pub.publish(cmplxNum)
        rate.sleep()



if __name__ == "__main__":
    try:
        pub_cmplx()
    except rospy.ROSInterruptException:
        pass
